src/backend/optimizer/plan/createplan.c

```diff
@@ -2026,8 +2028,8 @@ create_projection_plan(PlannerInfo *root, ProjectionPath *best_path, int flags)
 		 * Tell create_plan_recurse that we're going to ignore the tlist it
 		 * produces.
 		 */
-		subplan = create_plan_recurse(root, best_path->subpath,
-									  CP_IGNORE_TLIST);
+		int flags = is_function_scan_on_initplan(root, best_path->subpath) ? 0 : CP_IGNORE_TLIST;
```

Comment:
Maybe better to change name of `flags` to prevent the shadowing of the top-level variable?

Reply:
Done